hicommonwealth/commonwealth
# Fix refresh behavior + payload

## Link to Issue
Closes: #5653

## Description of Changes
- Applies refresh regardless of topic associations

## "How We Fixed It"
- Start with querying groups instead of topics

## Test Plan
- Add a group to the Ethereum community
  - Associate group with a few topics
- Make request: `PUT http://localhost:8080/api/refresh-membership`
  - Body `{
    "author_chain": "ethereum",
    "address": "<ADDRESS>,
    "chain": "ethereum",
    "jwt": "<JWT>"
}`
- Confirm that payload shows membership info including group ID and topic IDs

## Deployment Plan
N/A

## Other Considerations
Frontend will need to be updated wherever this route is used.

Example usage:
```ts
const refreshPayload = [
  {
    groupId: 574,
    topicIds: [3147, 96],
    allowed: false,
    rejectReason:
      '[{"requirement":{"rule":"threshold","data":{"threshold":"0","source":{"source_type":"eth_native","evm_chain_id":1}}},"message":"User Balance of 0 below threshold 0"}]',
  },
  {
    groupId: 575,
    topicIds: [],
    allowed: false,
    rejectReason:
      '[{"requirement":{"rule":"threshold","data":{"threshold":"100","source":{"source_type":"eth_native","evm_chain_id":1}}},"message":"User Balance of 0 below threshold 100"}]',
  },
];

function isAcceptedInTopic(topicId: number) {
  return !!refreshPayload.find(
    (entry) => entry.topicIds.includes(topicId) && entry.allowed
  );
}

function isAcceptedInGroup(groupId: number) {
  return !!refreshPayload.find(
    (entry) => entry.groupId === groupId && entry.allowed
  );
}
```
